h5py/_hl/dataset.py

```diff
@@ -409,13 +409,27 @@ def size(self):
             self._cache_props['size'] = size
         return size
 
+    @property
+    def _selector(self):
```

Comment:
There is a race condition here between two threads on with a read-only file, but I think it is mostly harmless (as the failure mode is we generate the selector multiple times and then future calls are cached.

Reply:
Yup, it should be no problem to have multiple selectors/readers on a single dataset. The cache is not to enforce that there's only one at once, it's just to avoid needlessly re-initialising one.

In fact, two threads should never be using a single selector/reader at the same time, because they're stateful objects. All the cases where a cached one could be used are protected by the `phil` lock.

Reply:
:+1: